### DISCUSSION TITLE
how to make LVGL work in Zephyr?

### DISCUSSION THREAD

**Discussion by glin00** [zephyrproject-rtos/zephyr#39002] [Q&A]
Description:
I am trying to enable LVGL inside Zephyr. After I add "CONFIG_LVGL=y", I try to call lv_init() within main() function but it can not work.  I have build the firmware without Zephyr OS with LVGL and LVGL is working correctly.  So, how  to make LVGL work in Zephyr?  Do I need to copy the whole LVGL source directory(see attached) into /src which is the way I did without Zephyr OS?

![image](https://user-images.githubusercontent.com/20088622/135434657-c0ee630c-c166-41a6-9bdf-f3809fbd48be.png)

Hardware: ST MCU STM32L496AGI6P, OLED with SPI interface
Zephyr Ver: Zephyr OS 2.7.0-rc2 
LVGL:v7.6.1
OS: Linux Ubuntu

Please advise me how to fix this problem. Thank You!

**Comment by glin00** [comment]
In order to debug in more detail, I have to setup a debugger in Linux. Does anybody know which debugger work the best using ST-LINK/V2?

**Reply by pdgendt** [reply]
Did you take a look at the sample `samples/subsys/display/lvgl`?
